src/kademlia/node.cpp

```diff
@@ -581,6 +581,7 @@ time_duration node::connection_timeout()
 	m_last_tracker_tick = now;
 
 	m_storage->tick();
+	update_dht_storage_counters();
```

Comment:
instead of sprinkling calls to this function anywhere the storage counters may be updated, you could just call it from:
session_impl::post_session_stats() (if DHT is enabled and if there is a dht object of course). you may have to thread the call through dht_tracker to get to the node. I think it's worth a try at least. I think the code would be more robust and a bit more efficient (since it's only called on-demand)